### ISSUE TITLE
U_sparsity throws an error when different cond.yz options are chosen for mra conditioning

### ISSUE THREAD

**Issue by marcinjurek** [katzfuss-group/GPvecchia#32]
```{R}
vecchia_specify(locs,m=3,cond.yz='SGV',conditioning='mra')
vecchia_specify(locs,m=3,cond.yz='y',conditioning='mra')
```

The first version with the default cond.yz='SGV' works, while the second version returns an error generated by line 47 in U_sparsity.R. This is weird, because cond.yz='SGV' and cond.yz='y' should be exactly equivalent in this setting, and as far as I could tell, both versions pass the exact same Cond object in vecchia_specify() to the U_sparsity() function.

**Comment by marcinjurek** [comment]
So the problem was caused by my algorithm which returned an NNarray that sometimes had fewer than m+1 columns. This happens when *effective m* is less than m which might happen for example if J=33. Then it is rounded to the next power of 2. In this case it would make a big difference. In practice effective m is rarely observed to be much less than m but if it's off even by 1, it used to throw off the algorithm. Fixed in 95995bb

**Comment by marcinjurek** [comment]
The problem was happening due to an error in my NNarray construction algorithm which led to arrays wit fewer than m columns. This happened, because for MRA conditioning *effective m* is sometimes slightly less than m. But the NNarray matrix is assumed throughout to have m columns so if effective m is less, I changed it in 95995bb so that now it adds the neccessary number of columns with NAs to ensure correct dimensions.

**Comment by katzfuss** [comment]
Is it possible to redefine m to reflect the actual, effective m, instead of padding NNarray with NAs? If so, please implement that way, to avoid unnecessary overhead due to extra columns in NNarray.

**Comment by marcinjurek** [comment]
Oh, it is possible and easy. However, I just wasn't sure if this assumption that NNarray is Nx(m+1) is not used somewhere else in the code. But seems like it isn't so no problem.

**Comment by marcinjurek** [comment]
fixed in 71f7179
